be/test/exec/hdfs_scanner_test.cpp

```diff
@@ -637,6 +637,100 @@ TEST_F(HdfsScannerTest, TestOrcGetNextWithDictFilter) {
 
 // ====================================================================================================
 
+/**
+ *
+File be/test/exec/test_data/orc_scanner/two-strips-dict-and-nodict.orc has 2 stripes
+Stripe Statistics:
+  Stripe 1:
+    Column 0: count: 100 hasNull: false
+    Column 1: count: 100 hasNull: false bytesOnDisk: 19 min: 0 max: 9 sum: 450
+    Column 2: count: 100 hasNull: false bytesOnDisk: 17 min: a max: a sum: 100
+  Stripe 2:
+    Column 0: count: 100 hasNull: false
+    Column 1: count: 100 hasNull: false bytesOnDisk: 20 min: 0 max: 9 sum: 451
+    Column 2: count: 100 hasNull: false bytesOnDisk: 1234 min: a max: ztrddxrhvzodrpeomoie sum: 1981
+
+File Statistics:
+  Column 0: count: 200 hasNull: false
+  Column 1: count: 200 hasNull: false bytesOnDisk: 39 min: 0 max: 9 sum: 901
+  Column 2: count: 200 hasNull: false bytesOnDisk: 1251 min: a max: ztrddxrhvzodrpeomoie sum: 2081
+
+Stripes:
+ Stripe1: offset: 3 data: 36 rows: 100 tail: 52 index: 63
+   Stream: column 0 section ROW_INDEX start: 3 length 11
+   Stream: column 1 section ROW_INDEX start: 14 length 25
+   Stream: column 2 section ROW_INDEX start: 39 length 27
+   Stream: column 1 section DATA start: 66 length 19
+   Stream: column 2 section DATA start: 85 length 7
+   Stream: column 2 section LENGTH start: 92 length 6
+   Stream: column 2 section DICTIONARY_DATA start: 98 length 4
+   Encoding column 0: DIRECT
+   Encoding column 1: DIRECT_V2
+   Encoding column 2: DICTIONARY_V2[1]
+ Stripe2: offset: 154 data: 1254 rows: 100 tail: 50 index: 83
+   Stream: column 0 section ROW_INDEX start: 154 length 11
+   Stream: column 1 section ROW_INDEX start: 165 length 25
+   Stream: column 2 section ROW_INDEX start: 190 length 47
+   Stream: column 1 section DATA start: 237 length 20
+   Stream: column 2 section DATA start: 257 length 1224
+   Stream: column 2 section LENGTH start: 1481 length 10
+   Encoding column 0: DIRECT
+   Encoding column 1: DIRECT_V2
+   Encoding column 2: DIRECT_V2
+*/
+
+TEST_F(HdfsScannerTest, TestOrcGetNextWithDiffEncodeDictFilter) {
+    SlotDesc two_diff_encode_orc_desc[] = {{"x", TypeDescriptor::from_logical_type(LogicalType::TYPE_BIGINT)},
```

Comment:
it's not BIGINT, it's INT.

the orc schema is

```
Type: struct<x:int,y:string>
```

Reply:
@dirtysalt  Thank you for your cr and suggestion, i got it.  I also appreciate the help from @Smith-Cruise